### DISCUSSION TITLE
Getting the line content for git blame

### DISCUSSION THREAD

**Discussion by dimiterbak** [community/community#24539] [Apps, API and Webhooks]
<p>How can I get the line content when runing “git blame”? Like if I use git blame on CLI I get the results in the format:</p>
<table><tbody>
<tr>
<td>Id</td>
<td>Author</td>
<td>Timestamp</td>
<td>Line Number</td>
<td>Line Content</td>
</tr>
<tr>
<td>89feb84d</td>
<td>Albert So</td>
<td>2018-03-01 00:54:03 +0000</td>
<td>3</td>
<td>This repository is an example of a project with multiple contributors making commits.</td>
</tr>
</tbody></table>
<p>I’d like to know what to add to the bellow “blame” query?</p>
<p>query blame {<br>
repository(name: “my_git-hammer”, owner: “dimiterbak”) {<br>
ref(qualifiedName: “master”) {<br>
target {<br>
… on Commit {<br>
blame(path: “azbg_digital_health_id-config.json”) {<br>
ranges {<br>
commit {<br>
oid<br>
author {<br>
name<br>
}<br>
commitResourcePath<br>
}<br>
startingLine<br>
endingLine<br>
age<br>
}<br>
}<br>
}<br>
}<br>
}<br>
}<br>
}</p>
<p>When I run it I get the bellow response:</p>
<p>{<br>
“data”: {<br>
“repository”: {<br>
“ref”: {<br>
“target”: {<br>
“blame”: {<br>
“ranges”: [<br>
{<br>
“commit”: {<br>
“oid”: “41f32d92ad849ac7e177c4001ea61055f1238d4c”,<br>
“author”: {<br>
“name”: “Dimitar”<br>
},<br>
“commitResourcePath”: “/dimiterbak/my_git-hammer/commit/41f32d92ad849ac7e177c4001ea61055f1238d4c”<br>
},<br>
“startingLine”: 1,<br>
“endingLine”: 26,<br>
“age”: 2<br>
}<br>
]<br>
}<br>
}<br>
}<br>
}<br>
}<br>
}</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/dimiterbak">@dimiterbak</a>, thank you for being here! As far as I know the only fields available are the ones you’ve already used on your query: startingLine, endingLine, author and age.</p>

**Comment by francisfuzz** [comment, marked as answer]
<p>Hello <a class="mention" href="/dimiterbak">@dimiterbak</a>! In addition to what <a class="mention" href="/AndreaGriffiths11">@AndreaGriffiths11</a> mentioned :sparkles: , I’d like to clarify that there isn’t a specific field (or field within an existing field) that would give you the line content beyond what’s available in a blame’s <code>blame range</code> object:</p>
<aside class="onebox whitelistedgeneric">
  <header class="source">
      <a href="https://developer.github.com/v4/object/blamerange/" target="_blank">GitHub Developer</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/15894826/181085211-f9249fe5-5b8c-424d-b847-af7d4a0ab4d7.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/26361d706a8a255479f0cb328577f99ad8731275_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/2/26361d706a8a255479f0cb328577f99ad8731275_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/2/26361d706a8a255479f0cb328577f99ad8731275_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/26361d706a8a255479f0cb328577f99ad8731275_2_10x10.png">

<h3><a href="https://developer.github.com/v4/object/blamerange/" target="_blank">BlameRange</a></h3>

<p>Get started with one of our guides, or jump straight into the API documentation.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Also, there’s not a specific REST API endpoint that exposes the same information. I hope this helps!</p>
